src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -265,197 +265,163 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
-                            reducedMethodSymbol.Name,
-                            reducedMethodSymbol.Arity,
-                            reducedMethodSymbol.GetGlyph(),
-                            qualifiedNamespaceName));
-                    }
+                if (cacheEntry == null)
+                {
+                    // Don't provide anything if we don't have all the required SyntaxTreeIndex created.
+                    return GetIndicesResult.NoneResult;
                 }
-            }
-        }
 
-        private static void GetItemsFromConflictingTypes(
-            INamespaceSymbol containingNamespaceSymbol,
-            ConflictNameNode conflictTypeNodes,
-            ArrayBuilder<SerializableImportCompletionItem> builder,
-            ITypeSymbol receiverTypeSymbol,
-            SemanticModel semanticModel,
-            int position,
-            StatisticCounter counter)
-        {
-            Debug.Assert(!conflictTypeNodes.NamespaceAndMethodNames.HasValue);
+                syntaxIndices.Add(project, cacheEntry.Value);
+            }
 
-            foreach (var child in conflictTypeNodes.Children.Values)
+            // Search through all direct PE references.
+            foreach (var peReference in currentProject.MetadataReferences.OfType<PortableExecutableReference>())
             {
-                if (child.NamespaceAndMethodNames == null)
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+                    solution, peReference, loadOnly: !forceIndexCreation, cancellationToken).ConfigureAwait(false);
+
+                if (info == null)
                 {
-                    var childNamespace = containingNamespaceSymbol.GetMembers(child.Name).OfType<INamespaceSymbol>().FirstOrDefault();
-                    if (childNamespace != null)
-                    {
-                        GetItemsFromConflictingTypes(childNamespace, child, builder, receiverTypeSymbol, semanticModel, position, counter);
-                    }
+                    // Don't provide anything if we don't have all the required SymbolTreeInfo created.
+                    return GetIndicesResult.NoneResult;
                 }
-                else
+
+                if (info.ContainsExtensionMethod)
                 {
-                    var types = containingNamespaceSymbol.GetMembers(child.Name).OfType<INamedTypeSymbol>();
-                    foreach (var type in types)
-                    {
-                        var (namespaceName, methodNames) = child.NamespaceAndMethodNames.Value;
-                        GetItemsFromTypeContainsPotentialMatches(type, namespaceName, methodNames, receiverTypeSymbol, semanticModel, position, counter, builder);
-                    }
+                    symbolInfos.Add(peReference, info);
                 }
             }
+
+            return new GetIndicesResult(syntaxIndices, symbolInfos);
```

Comment:
nit for the future.  the names syntaxIndicies and symbolInfos was very confusing when seeing it confused up above.  You might want something more like ProjectSourceInfo and MetadataInfo to make it clearer what each represented.

Reply:
will fix in the follow up PR